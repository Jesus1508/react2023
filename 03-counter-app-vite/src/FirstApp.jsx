import PropTypes from 'prop-types'

export const FirstApp = ( { title, subtitle, name } ) => {
    
    return (
        <>
            <h1 data-testid="test-title">{ title }</h1>
            <p>{ subtitle }</p>
            <p>{ subtitle }</p>
            <p>{ name }</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    name:PropTypes.string
}

FirstApp.defaultProps = {
    name: 'Jesus Solis Martinez',
    subtitle: 'No hay subtitulo',
    // title: 'No hay nigun titulo',
}