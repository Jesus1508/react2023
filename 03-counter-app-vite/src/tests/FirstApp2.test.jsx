import { render, screen } from '@testing-library/react'
import { FirstApp } from '../FirstApp';

/* eslint-disable no-undef */
describe('Pruebas sobre FirstApp Component', () => {
    const title = 'Hola, Soy Jesus'
    const subTitle = 'Soy un subtitulo'
    test('debe de hacer match con el snapshot', () => {
        
        const { container } = render (<FirstApp title={title}/>)

        expect( container ).toMatchSnapshot()
    });

    test('debe de mostrar el mensaje "Hola, Soy Jesus', () => {
        render (<FirstApp title={title}/>)
        expect( screen.getByText(title) ).toBeTruthy()
    });

    test('debde de mostrar el titulo en un h1', () => {
        render (<FirstApp title={title}/>)
        expect( screen.getByRole('heading', { level: 1}).innerHTML ).toContain( title )
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        render (
            <FirstApp 
                title={title}
                subtitle={subTitle}
            />
        )

        expect( screen.getAllByText(subTitle).length ).toBe(2)
    });
    
});