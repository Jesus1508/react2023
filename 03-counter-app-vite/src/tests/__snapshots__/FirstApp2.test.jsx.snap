// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre FirstApp Component debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Jesus
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Jesus Solis Martinez
  </p>
</div>
`;
